src/Nri/Ui/RadioButton/V4.elm

```diff
@@ -0,0 +1,639 @@
+                , pseudoClass "focus"
+                    [ Css.Global.adjacentSiblings
+                        [ Css.Global.everything
+                            [ Css.Global.descendants
+                                [ Css.Global.class "Nri-RadioButton-RadioButtonIcon"
+                                    [ borderColor (rgb 0 95 204)
+                                    ]
+                                ]
+                            ]
+                        ]
+                    ]
+                ]
+             ]
+                ++ List.map (Attributes.map never) config.custom
+            )
+         , Html.label
+            [ for idValue
+            , classList
+                [ ( "Nri-RadioButton-RadioButton", True )
+                , ( "Nri-RadioButton-RadioButtonChecked", isChecked )
+                ]
+            , css
+                [ outline Css.none
+                , Fonts.baseFont
+                , Css.batch
+                    (if config.isDisabled then
+                        [ color Colors.gray45
+                        , cursor notAllowed
+                        ]
+
+                     else if isInError then
+                        [ color Colors.purple
+                        , cursor pointer
+                        ]
+
+                     else
+                        [ color Colors.navy
+                        , cursor pointer
+                        ]
+                    )
+                , margin zero
+                , padding zero
+                , fontSize (px 15)
+                , Css.property "font-weight" "600"
+                , display inlineBlock
+                , Css.property "transition" "all 0.4s ease"
+                ]
+            ]
+            [ radioInputIcon
+                { isLocked = isLocked
+                , isDisabled = config.isDisabled
+                , isChecked = isChecked
+                }
+            , span
+                [ css
+                    [ display inlineFlex
+                    , alignItems center
+                    , Css.height (px 20)
+                    ]
+                ]
+                [ Html.span
+                    [ css <|
+                        if config.hideLabel then
+                            [ Css.width (px 1)
+                            , overflow Css.hidden
+                            , margin (px -1)
+                            , padding (px 0)
+                            , border (px 0)
+                            , display inlineBlock
+                            , textIndent (px 1)
+                            ]
+
+                        else
+                            config.labelCss
+                    ]
+                    [ Html.text label ]
+                , case ( config.premiumDisplay, config.premiumMsg ) of
+                    ( PremiumDisplay.Free, _ ) ->
+                        text ""
+
+                    ( _, Nothing ) ->
+                        text ""
```

Comment:
This is fine and can be changed later, but in the spirit of showing that an option is PremiumUnlocked, shouldn't we show the pennant if the config.premiumDisplay is PremiumDisplay.PremiumUnlocked or PremiumDisplay.PremiumLocked ? This would be a non clickable pennant.

Reply:
I believe it'd be better to introduce the visual changes already, since that also implies on API changes:
`showPennant`, for example, doesn't make sense if the pennant is always shown for premium content, a name like `onClickPennant` would make more sense.